src/lua/cdebugger.lua

```diff
@@ -0,0 +1,900 @@
+            info = nil
+            break
+        end
+        info = debug.getinfo(offset + CMD_STACK_LEVEL)
+    until frame_has_line(info)
+
+    if info then
+        stack_inspect_offset = offset
+        dbg_writeln("Inspecting frame: " .. format_stack_frame_info(info))
+        if tonumber(dbg.auto_where) then
+            where(info, dbg.auto_where)
+        end
+    else
+        -- info = debug.getinfo(stack_inspect_offset + CMD_STACK_LEVEL)
+        dbg_writeln("Already at the top of the stack.")
+    end
+
+    return false
+end
+
+local function cmd_where(context_lines)
+    local info = debug.getinfo(stack_inspect_offset + CMD_STACK_LEVEL, "Snl")
+    return (info and where(info, tonumber(context_lines) or 5))
+end
+
+local function cmd_listing(context_lines)
+    local offset = stack_inspect_offset + CMD_STACK_LEVEL - 2
+    local info = debug.getinfo(offset, "Snl")
+    return (info and where(info, tonumber(context_lines) or 5))
+end
+
+local function cmd_trace()
+    dbg_writeln("Inspecting frame %d", stack_inspect_offset - stack_top)
+    local i = 0;
+    while true do
+        local info = debug.getinfo(stack_top + CMD_STACK_LEVEL + i, "Snl")
+        if not info then break end
+
+        local is_current_frame = (i + stack_top == stack_inspect_offset)
+        local tab_or_caret = (is_current_frame and GREEN_CARET or "    ")
+        dbg_writeln(color_grey("% 4d") .. tab_or_caret .. "%s",
+                    i, format_stack_frame_info(info))
+        i = i + 1
+    end
+
+    return false
+end
+
+local function cmd_locals()
+    local bindings = local_bindings(1, false)
+
+    -- Get all the variable binding names and sort them
+    local keys = {}
+    for k, _ in pairs(bindings) do table.insert(keys, k) end
+    table.sort(keys)
+
+    for _, k in ipairs(keys) do
+        local v = bindings[k]
+
+        -- Skip the debugger object itself, "(*internal)" values, and Lua 5.2's
+        -- _ENV object.
+        if not rawequal(v, dbg) and k ~= "_ENV" and not k:match("%(.*%)") then
+            dbg_writeln("  " .. color_blue(k) .. pretty(v))
+        end
+    end
+
+    return false
+end
+
+local gen_commands
+
+local function cmd_help()
+    for k, map in pairs(gen_commands) do
+        if map.first then
+            if #map.aliases > 0 then
+                local fun = require 'fun'
+                local txt = ''
+                fun.each(function(x) txt = txt .. '|' .. color_yellow(x) end,
+                            map.aliases)
+                print(color_blue(k) .. '|' .. string.sub(txt, 2, #txt) ..
+                        ' ' .. (map.arg or ''))
+            else
+                print(color_blue(k) .. ' ' .. (map.arg or ''));
+            end
+            print('    - ' .. map.help)
+        end
+    end
+    return false
+end
+
+local function cmd_quit()
+    dbg.exit(0)
+    return true
+end
+
+local commands_help = {
+    --{'', '<return> to repeat last command', cmd_lastcommand},
```

Comment:
Because:
1. you could hardly create Lua table with empty string as a key;
2. and it's handled separately in https://github.com/tarantool/tarantool/pull/7593/files#diff-cac580d77550c2763ad0313654a750953f11e5dde3403b724d883640f8b36c8eR642

Reply:
👍